src/Tribe/Cost_Utils.php

```diff
@@ -253,12 +266,24 @@ public function parse_cost_range( $cost ) {
 			$cost = reset( $matches );
 		}
 
+		// Get the max number of decimals for the range
+		if ( count( $matches ) === 4 ) {
+			$decimals = max( array_map( 'strlen', end( $matches ) ) );
+
+			// If we passed max decimals
+			if ( ! is_null( $max_decimals ) ) {
+				$decimals = max( $max_decimals, $decimals );
+			}
+		}
+
 		$cost = (array) $cost;
 		$ocost = array();
 
 		// Keep the Costs in a organizeable array by keys with the "numeric" value
 		foreach ( $cost as $key => $value ) {
-			$ocost[ str_replace( $separators, '', $value ) ] = $value;
+			// Creates a Well Balanced Index that will perform good on a Key Sorting method
+			$index = str_replace( '.', '', number_format( str_replace( $separators, '.', $value ), $decimals ) );
```

Comment:
Realise this has already been merged, but on writing some tests for a separate cost-related issue I hit the problem that this will create an index of `599`for the value `5.99` ... since we compare the indices when finding the max/min costs, this can cause unexpected results if the highest cost is actually `100`, as of course `599 > 100`.